File: code/sorting/src/bubble_sort/bubble_sort.dart
void main() {
  List<int> array = [5, 1, 4, 2, 8];
  List<int> sortedarray = bubbleSort(array);
  print(sortedarray);
}

bubbleSort(List<int> array) {
  int lengthOfArray = array.length;
  for (int i = 0; i < lengthOfArray - 1; i++) {
    print('Index i at pos: ${i}');
    for (int j = 0; j < lengthOfArray - i - 1; j++) {
      print('loop:${i}');
      print('index i and j at pos: ${i}  &  ${j}');

      if (array[j] > array[j + 1]) {
        int temp = array[j];
        array[j] = array[j + 1];
        array[j + 1] = temp;
      }
    }
  }
  return (array);
}

File: code/languages/Java/Bubblesort.java
public class Bubblesort
{
   public static void printArray(int arr[]){
   for (int i = 0 ; i < arr.length ; i++){
      System.out.println(arr[i]+"");
      }
   }
   public static void main(String[] args) {
      int arr[] = {7 , 9 , 5 , 22 , 344 , 32, 5};
      {
         for (int i = 0 ; i < arr.length;i++)
         {
            for (int j = 0 ; j < arr.length - i - 1 ; j++)
            {
               if (arr[j] > arr[j+1]){
                  //swap
                  int temp = arr[j];
                  arr[j] = arr[j+1];
                  arr[j+1] = temp;
               }
            }
         }
      }
      printArray(arr);
   }
}
